apis/mapping/v1alpha1/namespaceoffloading_types.go

```diff
@@ -0,0 +1,96 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+)
+
+type mappingStrategy string
+
+const (
+	// EqualName -> remote Namespace will have the same name of local Namespace.
+	EqualName mappingStrategy = "EnforceSame"
+	// StandardName -> remote Namespace will have a default name: localNamespaceName-localClusterID.
+	StandardName mappingStrategy = "Default"
+)
+
+type deployingStrategy string
+
+const (
+	// LocalDeploy allows user only to deploy pod on local cluster.
+	LocalDeploy deployingStrategy = "Local"
+	// RemoteDeploy allows user only to deploy pod on remote clusters, possibly filtered through ClusterSelector.
+	RemoteDeploy deployingStrategy = "Remote"
+	// LocalAndRemoteDeploy allows user to deploy pod locally and remotely.
+	LocalAndRemoteDeploy deployingStrategy = "LocalAndRemote"
+)
+
+// NamespaceOffloadingSpec defines the desired state of NamespaceOffloading.
+type NamespaceOffloadingSpec struct {
+	// RemappingNameStrategy allows user to map remote Namespace name with 2 different strategy
+	// (Look at mappingStrategy).
+	// +kubebuilder:validation:Enum="EnforceSame";"Default"
+	// +kubebuilder:default="Default"
+	RemappingNameStrategy mappingStrategy `json:"remappingNameStrategy,omitempty"`
+
+	// If DeployingPodStrategy allows user to deploy pods in three different scenario:
+	// { locally, remotely, locally and remotely}
+	// +kubebuilder:validation:Enum="Local";"Remote";"LocalAndRemote"
+	// +kubebuilder:default="LocalAndRemote"
+	DeployingPodStrategy deployingStrategy `json:"deployingPodStrategy,omitempty"`
```

Comment:
```suggestion
	// PodOffloadingStrategy allows users to configure how pods in this namespace are offloaded, according to three different strategies: "Local" (i.e. no offloading is performed), "Remote" (i.e. all pods are offloaded in remote clusters), "LocalAndRemote" (i.e. no constraints are enforced besides the ones specified by the ClusterSelector):
	// { locally, remotely, locally and remotely}
	// +kubebuilder:validation:Enum="Local";"Remote";"LocalAndRemote"
	// +kubebuilder:default="LocalAndRemote"
	PodOffloadingStrategy podOfloadingStrategy `json:"deployingPodStrategy,omitempty"`
```

Same as befor about `omitempty`